packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -40,74 +51,92 @@ export function setResponsiveMode(responsiveMode: ResponsiveMode | undefined): v
 export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveMode }, TState>(
   ComposedComponent: new (props: TProps, ...args: any[]) => React.Component<TProps, TState>
 ): any {
-  const resultClass = class WithResponsiveMode extends BaseDecorator<TProps, IWithResponsiveModeState> {
-    constructor(props: TProps) {
-      super(props);
-      this._updateComposedComponentRef = this._updateComposedComponentRef.bind(this);
-
-      this.state = {
-        responsiveMode: _defaultMode || _lastMode || ResponsiveMode.large
-      };
-    }
-
-    public componentDidMount(): void {
-      this._events.on(window, 'resize', this._onResize);
-      this._onResize();
-    }
+  // tslint:disable-next-line:function-name
+  function WithResponsiveModeBase(
+    props: TProps & {
+      getWindow: () => Window | undefined;
+    },
+    forwardedRef: React.Ref<React.Component<TProps, TState>>
+  ): JSX.Element | null {
+    const [responsiveMode = ResponsiveMode.unknown, setCurrentResponsiveMode] = React.useState<ResponsiveMode>(
+      _defaultMode || _lastMode || ResponsiveMode.large
+    );
+
+    React.useLayoutEffect(() => {
```

Comment:
should be `React.useEffect`

Reply:
If I do that it a lot more unit tests will need a re-write, to wrap all logic with `renderer.act()`. Also, Enzyme doesn't do it synchronously, so we'd need a way to capture it.

Reply:
Since this hook is not commonly used, can you add a comment explaining why you're using it here?